env-builder/src/flox-env.cc

```diff
@@ -0,0 +1,309 @@
+            true,
+            package.priority );
+          references.insert( output.second.value() );
+        }
+
+      /* Collect drvs that may yet need to be built */
+      if ( auto drvPath = package_drv->queryDrvPath() )
+        {
+          drvsToBuild.push_back( { *drvPath } );
+        }
+    }
+
+  /* Build derivations that make up the environment */
+  // todo check if this builds `outputsToInstall` only
+  state.store->buildPaths( toDerivedPaths( drvsToBuild ),
+                           state.repair ? bmRepair : bmNormal );
+
+  // todo: is it script _xor_ file?
+  if ( auto hook = lockfile.getManifest().getManifestRaw().hook )
+    {
+      nix::Path script_path;
+
+      if ( auto script = hook->script )
+        {
+          script_path = createTempFile().second;
+          std::ofstream file( script_path );
+          file << script.value();
+          file.close();
+        }
+
+
+      if ( auto file = hook->file ) { script_path = file.value(); }
+
+      if ( ! script_path.empty() )
+        {
+
+          auto tempDir = createTempDir();
+          std::filesystem::copy_file( script_path,
+                                      std::filesystem::path( tempDir )
+                                        / "activation-hook.sh" );
+
+          auto script_store_path
+            = state.store->addToStore( "activation-hook.sh", tempDir );
+
+          references.insert( script_store_path );
+          pkgs.emplace_back( state.store->printStorePath( script_store_path ),
+                             true,
+                             0 );
+        }
+    }
+
+  /**
+   * insert activation script
+   */
+  auto activation_script_path
+    = state.store->parseStorePath( ACTIVATION_SCRIPT_BIN );
+
+  state.store->ensurePath( activation_script_path );
+  references.insert( activation_script_path );
+  pkgs.emplace_back( state.store->printStorePath( activation_script_path ),
+                     true,
+                     0 );
+
+  return createEnvironmentStorePath( state, pkgs, references );
+}
+
+
+struct CmdBuildEnv : nix::EvalCommand
+{
+  std::string              lockfile_content;
+  std::optional<nix::Path> out_link;
+
+  CmdBuildEnv()
+  {
+    // expectArgs( { .label = "lockfile", .handler = { &lockfile_content } } );
+
+    addFlag( { .longName    = "lockfile",
+               .shortName   = 'l',
+               .description = "locked manifest",
+               .labels      = { "lockfile" },
+               .handler     = { &lockfile_content } } );
+
+    addFlag( { .longName    = "out-link",
+               .shortName   = 'o',
+               .description = "output link",
+               .labels      = { "out-link" },
+               .handler     = { &out_link } } );
+  }
+
+  std::string
+  description() override
+  {
+    return "build flox env";
+  }
+
+  std::string
+  doc() override
+  {
+    return "TODO";
+  }
+
+
+  void
+  run( ref<Store> store ) override
```

Comment:
Lets print debug stuff to stderr and *only* store path of the just created flox environment (`store_path`) to the stdout.